### PR TITLE
Reordering categories propagates to legends and axes

### PR DESCRIPTION
**Pull request by bfinzer**
[#185325780] Feature: Category reordering propagates to legends and graph axes

* Separate storing all current colors for an attribute's categories from swapCategoriesForAttrRole since swapping by dragging axis categories does not involve colors.
* GraphController installs CategorySets in its MultiScales when appropriate
* MultiScale responds to category set values change by updating the scale band domain
* Rendering plots on category set change
* Properly disposing of MultiScale autorun

### CODE REVIEW

File: v3/src/components/axis/components/axis.tsx [hunk 1]
```diff
@@ -13,12 +14,14 @@ interface IProps {
   enableAnimation: MutableRefObject<boolean>
   showScatterPlotGridLines?: boolean
   centerCategoryLabels?: boolean
+  getCategorySet?: () => ICategorySet | undefined  // only used for categorical axes
```
[the remark this answers is not in the document]
Comment (kswenson) [reply v3/src/components/axis/components/axis.tsx]: Removed in a code review commit.

File: v3/src/components/graph/models/data-configuration-model.ts [hunk 2]
```diff
@@ -298,14 +304,40 @@ export const DataConfigurationModel = types
...
+          const temp = catIndex1
+          catIndex1 = catIndex2
+          catIndex2 = temp
+        }
+        if (categorySet && numCategories > catIndex1 && numCategories > catIndex2) {
+          const cat1 = categoryArray[catIndex1],
+            beforeCat = catIndex2 < numCategories - 1 ? categoryArray[catIndex2 + 1] : undefined
+          categorySet.move(cat1, beforeCat)
+        }
+      },
```
[the remark this answers is not in the document]
Comment (kswenson) [reply v3/src/components/graph/models/data-configuration-model.ts]: Done in a code review commit.

**Review by kswenson — APPROVED** [review]
👍 Looks good -- added a code review commit to rebase to main and tweaks a couple things. Just a couple questions remaining.

File: v3/src/components/axis/hooks/use-sub-axis.ts [hunk 3]
```diff
@@ -166,6 +167,19 @@ export const useSubAxis = ({
     }
   }, [isNumeric, axisModel, refreshSubAxis, layout])
 
+  // Refresh when category set, if any, changes
+  useEffect(function installCategorySetSync() {
+    if (isCategorical) {
+      const disposer = autorun(() => {
+        const values = layout.getAxisMultiScale(axisModel.place)?.categorySetValues
+        if (values?.length) {
```
[the remark this answers is not in the document]
Comment (bfinzer) [reply v3/src/components/axis/hooks/use-sub-axis.ts:175]: It appears there are deeper problems that I'll log as a bug. (Deleting the last value doesn't trigger this autorun at all and then adding another category doesn't behave properly either.) But I think this bug belongs in another thread than this PR.

**Pull request merged**